sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -150,6 +154,49 @@
  *     ...
  * }</pre>
  *
+ * <p>Kafka provides deserializers for common types in
+ * {@link org.apache.kafka.common.serialization}. Alternatively, to use Kafka deserializers
+ * based on wrapping Beam {@link Coder}s, use {@code readWithCoders} instead of {@code read}:
+ *
+ * <pre>{@code
+ *
+ *  pipeline
+ *    .apply(KafkaIO.<MyKey, MyValue>readWithCoders(MyKeyCoder.of(), MyValueCoder.of())
+ *       .withBootstrapServers("broker_1:9092,broker_2:9092")
+ *       .withTopic("my_topic")
+ *    )
+ *    ...
+ * }</pre>
+ *
+ * <p>The key and value {@link Coder}s for the resulting collection are inferred from the
+ * deserializer types through the {@link Coder} registry. When the registry cannot infer the
+ * coder types, either register custom {@link Coder}s in the registry, or use
```

Comment:
This comment is confusing: you're still allowed to explicitly specify .withKeyCoder() and .withValueCoder(), right? I'd recommend to phrase this paragraph simply as:

In most cases, there's no need to specify key or value coders because they will be inferred from the respective deserializer. However, if that fails, you can specify them explicitly using withKeyCoder/withValueCoder().